src/locale/localed-util.c

```diff
@@ -697,6 +695,42 @@ int vconsole_convert_to_x11(const VCContext *vc, X11Context *ret) {
 
                 return x11_context_copy(ret, &xc);
         }
+
+        /* No custom mapping has been found, see if the keymap is a converted one. In such case deducing the
+         * corresponding x11 layout is easy. */
+        _cleanup_free_ char *xlayout = NULL, *converted = NULL;
+        char *xvariant;
+
+        xlayout = strdup(vc->keymap);
+        if (!xlayout)
+                return -ENOMEM;
+        xvariant = strchr(xlayout, '-');
+        if (xvariant) {
+                xvariant[0] = '\0';
+                xvariant++;
+        }
+
+        /* Note: by default we use keyboard model "microsoftpro" which should be equivalent to "pc105" but
+         * with the internet/media key mapping added. */
+        xc = (X11Context) {
+                .layout  = xlayout,
+                .model   = (char*) "microsoftpro",
+                .variant = xvariant,
+                .options = (char*) "terminate:ctrl_alt_bksp",
+        };
+
+        /* This sanity check seems redundant with the verification of the X11 layout done on the next
+         * step. However xkbcommon is an optional dependency hence the verification might be a NOP.  */
+        r = find_converted_keymap(&xc, &converted);
```

Comment:
Is it not necessary to check that `converted` is equivalent to `vc->keymap` ?

Reply:
It shouldn't be necessary unless `find_converted_keymap()` is buggy. We could add an `assert_se()` to check it but that would slightly complexify the code, not sure whether it's worth it...

Reply:
I mean, `assert(streq_ptr(converted, vc->keymap));`. I think that is not complicated.

Reply:
well it depends on whether `find_converted_keymap()` could have found a keymap which has the variant...

Something like this ?

```
diff --git a/src/locale/localed-util.c b/src/locale/localed-util.c
index 02fac9786b..a67f65ecb8 100644
--- a/src/locale/localed-util.c
+++ b/src/locale/localed-util.c
@@ -722,10 +722,14 @@ int vconsole_convert_to_x11(const VCContext *vc, X11Context *ret) {
         /* This sanity check seems redundant with the verification of the X11 layout done on the next
          * step. However xkbcommon is an optional dependency hence the verification might be a NOP.  */
         r = find_converted_keymap(&xc, &converted);
+        if (r > 0)
+                assert_se(streq(vc->keymap, converted));
         if (r == 0 && xc.variant) {
                 /* If we still haven't find a match, try with no variant, it's still better than nothing.  */
                 xc.variant = NULL;
                 r = find_converted_keymap(&xc, &converted);
+                if (r > 0)
+                        assert_se(startswith(vc->keymap, converted));
         }
         if (r < 0)
                 return r;

```

Reply:
Yeah. Something like that. But, after reconsidering that, `assert_se()` may be too strong. How about to log something and refuse the conversion?
```
if (!streq(vc->keymap, converted)) {
        log_debug(...);
        *ret = (X11Context) {};
        return 0;
}
```

Reply:
honestly I don't see the point of such sanity check. The way `find_converted_keymap()` is to find a converted keymap whose filename matches the passed xlayout hence it's already ensured by the function itself.

Also why would we add such check here and not at all other call sites to `find_converted_keymap()` as well ?